### PR TITLE
Add jar/zip support

### PR DESCRIPTION
**Pull request by showpune**
- [ ] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as \#\<issue\_number\>.
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

**Review by cormacpayne — APPROVED** [review]

**Review by pauld-msft — APPROVED** [review]

File: builders/container-apps-internal-registry-demo/Dockerfile [hunk 1]
```diff
@@ -38,6 +38,7 @@ RUN set -eux; \
 		sudo \
 		curl \
 		unzip \
+		grep \
```
Comment (yilims) [inline builders/container-apps-internal-registry-demo/Dockerfile:41]: grep should be already installed, no need to install here

File: builders/container-apps-internal-registry-demo/startup-script.sh [hunk 2]
```diff
@@ -28,18 +29,35 @@ if [ -n "$CORRELATION_ID" ]; then
...
   echo "Waiting for app source to be uploaded. Please upload the app source to the endpoint specified in the Build resource's 'uploadEndpoint' property."
-  curl -H "$auth_header" -H "$version_header" -H "$date_header" -X GET "$file_upload_endpoint" -o "$temp_app_source_path" -s
+  curl -H "$auth_header" -H "$version_header" -H "$date_header" -X GET "$file_upload_endpoint" -o "$temp_app_source_path" -D "$temp_app_header_path" -s
   sleep 5
+ 
+  if [[ -f "$temp_app_header_path" ]]; then
+    content_type=$(grep -i Content-Type "$temp_app_header_path" | cut -d ' ' -f2)
+  else 
+    echo ""$temp_app_header_path" not found"
+  fi
```
Comment (yilims) [inline builders/container-apps-internal-registry-demo/startup-script.sh:47]: how about move the code to line 50? or "$temp_app_header_path not found" will be printed in console serveral times.

File: builders/container-apps-internal-registry-demo/Dockerfile (hunk 1 above)
Comment (yilims) [inline builders/container-apps-internal-registry-demo/Dockerfile:41]: grep should be already installed, no need to install here.

**Review by yilims — CHANGES_REQUESTED** [review]

File: builders/container-apps-internal-registry-demo/startup-script.sh [hunk 3]
```diff
@@ -28,18 +29,33 @@ if [ -n "$CORRELATION_ID" ]; then
...
 done
 
 # Extract app code to CNB_APP_DIR directory.
-echo "Found app source at '$temp_app_source_path'. Extracting to $CNB_APP_DIR"
+echo "Found app source at '$temp_app_source_path' with format '$content_type'. Extracting to $CNB_APP_DIR"
 mkdir -p $CNB_APP_DIR
 cd $CNB_APP_DIR
-tar -xzf "$temp_app_source_path"
+
+if [[ $content_type =~ "application/zip" || $content_type =~ "java-archive" ]]; then
```
Comment (pauld-msft) [inline builders/container-apps-internal-registry-demo/startup-script.sh:53]: content type mistmatch between here and `application/java-archive` above

**Review by pauld-msft — APPROVED** [review]

**Review by yilims — APPROVED** [review]

File: builders/container-apps-internal-registry-demo/startup-script.sh [hunk 4]
```diff
@@ -25,21 +26,41 @@ done
...
+  echo -n "$CORRELATION_ID" > "$build_env_dir/CORRELATION_ID"
 fi
 
+file_extension=""
 while [[ ! -f "$temp_app_source_path" || ! "$(file $temp_app_source_path)" =~ "compressed data" ]]
 do
   echo "Waiting for app source to be uploaded. Please upload the app source to the endpoint specified in the Build resource's 'uploadEndpoint' property."
-  curl -H "$auth_header" -H "$version_header" -H "$date_header" -X GET "$file_upload_endpoint" -o "$temp_app_source_path" -s
+  curl -H "$auth_header" -H "$version_header" -H "$date_header" -X GET "$file_upload_endpoint" -o "$temp_app_source_path" -D "$temp_app_header_path" -s
   sleep 5
```
Comment (pauld-msft) [inline builders/container-apps-internal-registry-demo/startup-script.sh:37]: move sleep to after the file extension check

**Review by pauld-msft — APPROVED** [review]
minor comment, other than that looks good

**Review by pauld-msft — APPROVED** [review]

**Pull request merged**